plexus-compiler-manager/src/main/java/org/codehaus/plexus/compiler/manager/DefaultCompilerManager.java

```diff
@@ -36,19 +37,27 @@
 @Named
 public class DefaultCompilerManager implements CompilerManager {
     @Inject
-    private Map<String, Compiler> compilers;
+    private Map<String, Provider<Compiler>> compilers;
 
     // ----------------------------------------------------------------------
     // CompilerManager Implementation
     // ----------------------------------------------------------------------
 
     public Compiler getCompiler(String compilerId) throws NoSuchCompilerException {
-        Compiler compiler = compilers.get(compilerId);
+        // Provider<Class> is lazy
+        // presence of provider means component is present (but not yet constructed)
+        Provider<Compiler> compiler = compilers.get(compilerId);
 
+        // it exists: as provider is found
```

Comment:
Should that not mean that it is NOT there if it is null?

Reply:
yes, yes, bad worded comments but i hope the intent is clear :smile:

Reply:
Maybe remove or move down into the if case (code speaks for itself?)